src/caffe/proto/caffe.proto

```diff
@@ -30,6 +30,10 @@ message Datum {
   optional bool encoded = 7 [default = false];
 }
 
+message DatumVector {
```

Comment:
This is added because it allows one to define databases that store several Datums to one keyword.